blockchain-albatross/src/blockchain.rs

```diff
@@ -1062,12 +1062,12 @@ impl<'env> Blockchain<'env> {
         self.chain_store.get_blocks(start_block_hash, count, include_body, direction, None)
     }
 
-    pub fn get_transactions_root(&self, epoch: u32, txn_option: Option<&Transaction>) -> Option<Blake2bHash> {
-        let mut hashes = Vec::new();
+    pub fn get_epoch_transactions<B, F: Fn(&BlockchainTransaction) -> B>(&self, epoch: u32, f: F, txn_option: Option<&Transaction>) -> Option<Vec<B>> {
```

Comment:
The method name doesn't really indicate that you'll have to provide a closure to map the transactions. Maybe return an `impl Iterator<Item=&Transaction>`:

```rust
let iter_first = first_block.transactions()?.iter();
let iter_tail = blocks.into_iter().map(|block| block.transactions.iter()).flatten()
iter_first.chain(iter_tail)
```
Although I think this is not quite possible as we need to get the transactions by ownership and use `into_iter`. This can be achieved with:

```rust
if let Block::Micro(block) = block { block.transactions.into_iter() } else { return None }
```

Alternative is to just return the `Vec` of `Transaction`s. We get the blocks by ownership anyway.

Anyway, the consumer of this method should call `map` on it, instead of passing a closure. But I'm okay with keeping it this way, just to save time.